airflow/cli/api/client.py

```diff
@@ -0,0 +1,264 @@
+        if self.token:
+            request.headers["Authorization"] = "Bearer " + self.token
+        yield request
+
+
+class Client(httpx.Client):
+    """Client for the Airflow REST API."""
+
+    def __init__(self, *, base_url: str, token: str, **kwargs: Any):
+        auth = BearerAuth(token)
+
+        kwargs["base_url"] = f"{base_url}/public"
+        pyver = f"{'.'.join(map(str, sys.version_info[:3]))}"
+        super().__init__(
+            auth=auth,
+            headers={"user-agent": f"apache-airflow-cli/{version} (Python/{pyver})"},
+            event_hooks={"response": [raise_on_4xx_5xx], "request": [add_correlation_id]},
+            **kwargs,
+        )
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def assets(self):
+        """Operations related to assets."""
+        return AssetsOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def backfills(self):
+        """Operations related to backfills."""
+        return BackfillsOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def configs(self):
+        """Operations related to configs."""
+        return ConfigOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def connections(self):
+        """Operations related to connections."""
+        return ConnectionsOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def dags(self):
+        """Operations related to DAGs."""
+        return DagOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def dag_runs(self):
+        """Operations related to DAG runs."""
+        return DagRunOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def jobs(self):
+        """Operations related to jobs."""
+        return JobsOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def pools(self):
+        """Operations related to pools."""
+        return PoolsOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def providers(self):
+        """Operations related to providers."""
+        return ProvidersOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def variables(self):
+        """Operations related to variables."""
+        return VariablesOperations(self)
+
+    @lru_cache()  # type: ignore[misc]
+    @property
+    def version(self):
+        """Get the version of the server."""
+        return VersionOperations(self)
+
+
+# API Client Decorator for CLI Actions
+@contextlib.contextmanager
+def get_client():
+    """Get CLI API client."""
+    cli_api_client = None
+    try:
+        credentials = Credentials().load()
+        limits = httpx.Limits(max_keepalive_connections=1, max_connections=1)
+        cli_api_client = Client(base_url=credentials.api_url, limits=limits, token=credentials.api_token)
+        yield cli_api_client
+    except AirflowNotFoundException as e:
+        raise e
```

Comment:
Why catch then immediately reraise?

(I see this in a number of places, only commenting once)

Reply:
The idea was to leave CLI to decide on how to handle the error and log the proper error for the user. I first logged each error here and removed them to give to the CLI layer fully. I may add them back in a debug state, otherwise, the errors or warnings between CLI and there could be duplicates or confusing sometimes

Reply:
> The idea was to leave CLI to decide on how to handle the error and log the proper error for the user.

Makes sense.

This just stood out to me as, "oh, we forgot to do something". It also made me question if there was some esoteric behavior or feature I didn't know about. But, this can also be cleaned up in a follow up.